sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRelBoundedVsBoundedTest.java

```diff
@@ -227,6 +229,68 @@ public void testException_nonEqualJoin() throws Exception {
     pipeline.run();
   }
 
+  @Test
+  public void testException_join_condition1() throws Exception {
+    String sql =
+        "SELECT *  "
+            + "FROM ORDER_DETAILS1 o1"
+            + " JOIN ORDER_DETAILS2 o2"
+            + " on "
+            + " o1.order_id = o2.site_id OR o1.price = o2.site_id";
+
+    thrown.expect(UnsupportedOperationException.class);
+    thrown.expectMessage("Operator OR is not supported in join condition");
```

Comment:
I think general recommendation is not to compare the whole message, but rather that it contains things. Otherwise it becomes too brittle and harder to maintain. E.g. I think in this case it will be enough that it's an `UnsupportedException` with a message that contains `"operator OR"`

Reply:
Same in other messages